genie-agent/src/main/java/com/netflix/genie/agent/cli/JobRequestArgumentsImpl.java

```diff
@@ -64,10 +64,10 @@
     private boolean interactive;
 
     @Parameter(
-        names = {"--archivalDisabled"},
-        description = "Disable server-side archival of logs (does not disable logs upload and short-term retrieval)"
+        names = {"--archiveLocationPrefix"},
```

Comment:
Another possibility is to make this explicitly S3 specific, since that's what we support anyway right now.
i.e. `--s3ArchivalPrefix`.

Reply:
The problem is that there is already a field called `archiveLocation` in the db. This will store the actual archiveLocation. For e.g if the user passes in `s3://bucket/base` as the `requestedArchiveLocationPrefix`(s3 uri is just an example), then archiveLocation will store `s3://bucket/base/{jobid}`. Now, if I store `s3://bucket/base` in `requestedArchiveLocation` and `s3://bucket/base/{jobid}` in `archiveLocation` then the meaning of archiveLocation changes. One is really just the root uri and the other is the location of the job folder as it has `{jobId}` in the uri. By using `prefix` I was trying to make that clear. Hence I decided to call the field in the db `requestedArchiveLocationPrefix` and thinking backwards from there called the cli parameter with the same name. If `prefix` sounds s3 specific I can call the cli parameter `requestArchiveLocationRoot`.

Reply:
I can add a validator as you suggested and throw an error if its not a s3 uri. I would rather keep this parameter flexible as far as name goes so that tomorrow we can use it for other archival locations and simply relax the constraint in the validator.

Reply:
Here is the issue. Lets say we name it `s3ArchivalPrefix` and kragle or some other library which have migrated to the agent is making the call to the agent passing this option `s3ArchivalPrefix`. Then we rename it. Then those libraries will have to be changed. It seems unnecessary for them to do this. Hence I would like to stick to a more generic name `archiveLocationPrefix`. Since there is a validator now, non S3 urls will anyways fail. The description also says that currently only S3 is supported. This keeps things a little more flexible.